docs/managing-fork.md

```diff
@@ -0,0 +1,152 @@
+# Managing kubespray fork
+
+This document details the process to upgrade and/or make changes to the kubespray fork.
+
+TLDR:
+
+We only create one release branch `release-X.Y.Z-ck8s` in our fork per upstream release `X.Y.Z`.
+We then tag the latest commit of that release branch with `vX.Y.Z-ck8s<patch-number>`.
+
+## The fork is a git submodule
+
+`kubespray` is included in the `compliantkubernetes-kubespray` repo as a Git submodule, which means the only thing stored in `compliantkubernetes-kubespray` is metadata (path, url) in `.gitmodules` and a git tree object with a commit SHA-1 of where the submodule (our fork) is tracked.
+
+### Make sure the submodule is initialized and up to date
+
+Inspect the SHA-1 of the commit the superproject expects the fork working directory to be at:
+
+```sh
+git ls-tree <branch> kubespray
+```
+
+Inspect the SHA-1 of the currently checked out commit in the fork, along
+with the submodule path and the output of git describe for the SHA-1:
+
+```sh
+git submodule status
+```
+
+Make the local fork match what the remote superproject expects the directory to look like:
+
+```sh
+git submodule update --init
+```
+
+## Modifications of the fork
+
+### Step 1
+
+Make sure the fork and its remotes are set up correctly:
+
+```console
+$ cd kubespray
+
+$ git remote -v # make sure lines match
+  origin    ssh://git@github.com/elastisys/kubespray (fetch)
+  origin    ssh://git@github.com/elastisys/kubespray (push)
+  upstream  ssh://git@github.com/kubernetes-sigs/kubespray.git (fetch)
+  upstream  ssh://git@github.com/kubernetes-sigs/kubespray.git (push)
+
+# If not:
+$ git remote add upstream ssh://git@github.com/kubernetes-sigs/kubespray.git
+```
+
+### Step 2
+
+Make sure the master branch is up to date with the latest upstream release version:
+
+```console
+$ cd kubespray
+$ git fetch --all
+$ git switch master
+
+$ git status # Ensure these lines match
+On branch master
+Your branch is up to date with 'origin/master'.
```

Comment:
So if the output match I should stop and proceed to the step 3?

Reply:
yes, should i make it more clear?

Reply:
I would say yes if possible :)